### ISSUE TITLE
Better handling of insufficient posting privileges

### ISSUE THREAD

**Issue by daniloercoli** [wordpress-mobile/WordPress-Android#1296]
The app works OK with Editor' or 'Administrator' roles, and seems that it doesn't correctly handle XML-RPC fault code 401 ("Sorry, you cannot edit posts/comments/pages on the site"), and the fault code 403 ("You are not allowed access to details about this site.").

Instead, a generic error message - "Posts can't be refreshed at this time" - is displayed in the app.

**Comment by bummytime** [comment]
Closing issue and migrating to new repo.

**Comment by aforcier** [comment]
Re-opening this issue as it's more of a post list issue than an editor issue.

Relevant comment by @rachelmcr:

I tested with a Contributor user on both a WordPress.com site and a self-hosted site. In most cases, the app handles the permissions issues with understandable error messages.

In the editor, when I try to add media the media picker opens and a toast message appears, "You don't have permission to view the media library." A more persistent error message in the media picker might be helpful (similar to the errors currently in the Pages and Comments sections of the app), but this worked.

We could definitely use better handling around publishing posts, though. Would it be possible to remove the option to publish a post when the user doesn't have those privileges? As a Contributor user, I tried to publish a post and received an "Upload failed" notification with the message, "Post Upload failed: an error occurred while uploading" (not clear). The post was then listed as a local draft in the post list.
